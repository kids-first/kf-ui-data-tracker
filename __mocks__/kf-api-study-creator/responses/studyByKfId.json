{
  "data": {
    "studyByKfId": {
      "id": "U3R1ZHlOb2RlOlNEXzhXWDhRUTA2",
      "name": "benchmark extensible e-business",
      "shortName": null,
      "bucket": "stay-wrong-ready",
      "kfId": "SD_8WX8QQ06",
      "createdAt": "2019-04-20T15:22:18.780221+00:00",
      "modifiedAt": "2019-04-25T15:22:18.780221+00:00",
      "attribution": null,
      "dataAccessAuthority": "dbGaP",
      "externalId": "test_study_external_id",
      "releaseStatus": null,
      "version": null,
      "releaseDate": "2019-08-24",
      "description": "Sudy description in markdown, commonly the X01 abstract text. --  As test study description",
      "anticipatedSamples": 100,
      "awardeeOrganization": "CHOP",
      "collaborators": {
        "edges": [
          {
            "node": {
              "id": "VXNlck5vZGU6NQ==",
              "username": "Jennifer Warren",
              "firstName": "Marc",
              "lastName": "Wilson",
              "email": "ericksontracy@benitez-ruiz.com",
              "picture": "",
              "roles": [],
              "__typename": "UserNode"
            },
            "__typename": "UserNodeEdge"
          },
          {
            "node": {
              "id": "VXNlck5vZGU6OQ==",
              "username": "Anthony Hamilton",
              "firstName": "John",
              "lastName": "Garza",
              "email": "richard06@merritt.net",
              "picture": "",
              "roles": [],
              "__typename": "UserNode"
            },
            "__typename": "UserNodeEdge"
          },
          {
            "node": {
              "id": "VXNlck5vZGU6MjY=",
              "username": "devadmin",
              "firstName": "",
              "lastName": "",
              "email": "admin@myproject.com",
              "picture": "",
              "roles": ["ADMIN"],
              "__typename": "UserNode"
            },
            "__typename": "UserNodeEdge"
          }
        ],
        "__typename": "UserNodeConnection"
      },
      "files": {
        "edges": [
          {
            "node": {
              "id": "RmlsZU5vZGU6U0ZfNVpQRU0xNjc=",
              "kfId": "SF_5ZPEM167",
              "name": "organization.jpeg",
              "fileType": "CLN",
              "tags": [],
              "description": "Month necessary animal end standard case. View system operation message decade. Actually sing because deal everything woman subject.",
              "downloadUrl": "http://localhost:8080/download/study/SD_8WX8QQ06/file/SF_5ZPEM167",
              "versions": {
                "edges": [
                  {
                    "node": {
                      "id": "T2JqZWN0Tm9kZTpGVl80TlpOQkVPRg==",
                      "kfId": "FV_NVV741M1",
                      "state": "PEN",
                      "size": 9893,
                      "createdAt": "2018-08-21T03:16:11+00:00",
                      "__typename": "VersionNode"
                    },
                    "__typename": "VersionNodeEdge"
                  },
                  {
                    "node": {
                      "id": "T2JqZWN0Tm9kZTpGVl83NkhONTdaNw==",
                      "kfId": "FV_KCDKCDYR",
                      "state": "PEN",
                      "size": 5460,
                      "createdAt": "2018-09-27T05:32:26+00:00",
                      "__typename": "VersionNode"
                    },
                    "__typename": "VersionNodeEdge"
                  },
                  {
                    "node": {
                      "id": "T2JqZWN0Tm9kZTpGVl9IUkFHWE5XMQ==",
                      "kfId": "FV_800JI8IP",
                      "state": "PEN",
                      "size": 5419,
                      "createdAt": "2017-08-25T00:05:30+00:00",
                      "__typename": "VersionNode"
                    },
                    "__typename": "VersionNodeEdge"
                  }
                ],
                "__typename": "VersionNodeConnection"
              },
              "__typename": "FileNode"
            },
            "__typename": "FileNodeEdge"
          },
          {
            "node": {
              "id": "RmlsZU5vZGU6U0ZfWTA3SU4xSE8=",
              "kfId": "SF_Y07IN1HO",
              "name": "its.mp3",
              "fileType": "SHM",
              "tags": ["dbGaP", "random", "tag that has ten more char"],
              "description": "Question meeting move recognize.",
              "downloadUrl": "http://localhost:8080/download/study/SD_8WX8QQ06/file/SF_Y07IN1HO",
              "versions": {
                "edges": [
                  {
                    "node": {
                      "id": "T2JqZWN0Tm9kZTpGVl9UWjVROEhYWA==",
                      "kfId": "FV_IJ2CN4P9",
                      "state": "PEN",
                      "size": 7394,
                      "createdAt": "2017-08-22T08:32:38+00:00",
                      "__typename": "VersionNode"
                    },
                    "__typename": "VersionNodeEdge"
                  },
                  {
                    "node": {
                      "id": "T2JqZWN0Tm9kZTpGVl9aOUM1SUUwRg==",
                      "kfId": "FV_DNTN03CS",
                      "state": "PEN",
                      "size": 5883,
                      "createdAt": "2017-09-02T21:02:35+00:00",
                      "__typename": "VersionNode"
                    },
                    "__typename": "VersionNodeEdge"
                  },
                  {
                    "node": {
                      "id": "T2JqZWN0Tm9kZTpGVl9ITDg0SDhHTw==",
                      "kfId": "FV_ZDB8JOCT",
                      "state": "PEN",
                      "size": 1899,
                      "createdAt": "2019-03-14T05:46:35+00:00",
                      "__typename": "VersionNode"
                    },
                    "__typename": "VersionNodeEdge"
                  },
                  {
                    "node": {
                      "id": "T2JqZWN0Tm9kZTpGVl9YRVg4TERHMw==",
                      "kfId": "FV_10SBI6VA",
                      "state": "PEN",
                      "size": 7631,
                      "createdAt": "2018-03-13T23:52:08+00:00",
                      "__typename": "VersionNode"
                    },
                    "__typename": "VersionNodeEdge"
                  },
                  {
                    "node": {
                      "id": "T2JqZWN0Tm9kZTpGVl9HV1JLUUVDQQ==",
                      "kfId": "FV_8KF1VMN9",
                      "state": "PEN",
                      "size": 6834,
                      "createdAt": "2019-04-06T07:51:17+00:00",
                      "__typename": "VersionNode"
                    },
                    "__typename": "VersionNodeEdge"
                  }
                ],
                "__typename": "VersionNodeConnection"
              },
              "__typename": "FileNode"
            },
            "__typename": "FileNodeEdge"
          }
        ],
        "__typename": "FileNodeConnection"
      },
      "events": {
        "edges": [
          {
            "node": {
              "id": "RXZlbnROb2RlOjM5",
              "uuid": "b790803e-580e-458c-8b42-78252a5403f6",
              "createdAt": "2019-08-26T18:25:02.985155+00:00",
              "eventType": "FV_CRE",
              "description": "devadmin created version FV_9AF7CC3Z of file SF_5ZPEM167",
              "__typename": "EventNode"
            },
            "__typename": "EventNodeEdge"
          },
          {
            "node": {
              "id": "RXZlbnROb2RlOjM4",
              "uuid": "6372f6f7-8556-4b1c-9c29-7c0a7923f7ca",
              "createdAt": "2019-08-26T18:24:46.341665+00:00",
              "eventType": "FV_UPD",
              "description": "devadmin updated version FV_4FCEYZ3T of file SF_5ZPEM167",
              "__typename": "EventNode"
            },
            "__typename": "EventNodeEdge"
          },
          {
            "node": {
              "id": "RXZlbnROb2RlOjM2",
              "uuid": "ee949b92-ddb0-491b-b807-8f754e1641ff",
              "createdAt": "2019-08-26T18:24:36.763314+00:00",
              "eventType": "FV_CRE",
              "description": "devadmin created version FV_ABCPEQRK of file SF_Y07IN1HO",
              "__typename": "EventNode"
            },
            "__typename": "EventNodeEdge"
          },
          {
            "node": {
              "id": "RXZlbnROb2RlOjM1",
              "uuid": "05379686-3d68-4fed-b6a5-a5e47d97996f",
              "createdAt": "2019-08-26T18:24:19.986588+00:00",
              "eventType": "FV_CRE",
              "description": "devadmin created version FV_BHXN2F7A of file SF_Y07IN1HO",
              "__typename": "EventNode"
            },
            "__typename": "EventNodeEdge"
          },
          {
            "node": {
              "id": "RXZlbnROb2RlOjM0",
              "uuid": "1ec8bb79-b6ad-47eb-bd40-7ffc81135abe",
              "createdAt": "2019-08-26T18:24:03.184513+00:00",
              "eventType": "FV_UPD",
              "description": "devadmin updated version FV_Z0NZQZNX of file SF_Y07IN1HO",
              "__typename": "EventNode"
            },
            "__typename": "EventNodeEdge"
          },
          {
            "node": {
              "id": "RXZlbnROb2RlOjMz",
              "uuid": "b9b6bf82-1bdc-4d79-b69b-5e0350d64e52",
              "createdAt": "2019-08-26T18:24:03.182570+00:00",
              "eventType": "SF_UPD",
              "description": "devadmin updated file SF_5ZPEM167",
              "__typename": "EventNode"
            },
            "__typename": "EventNodeEdge"
          },
          {
            "node": {
              "id": "RXZlbnROb2RlOjMy",
              "uuid": "02564a4f-ead8-4ebb-af12-7171dcc05fd4",
              "createdAt": "2019-08-26T18:23:53.703031+00:00",
              "eventType": "FV_CRE",
              "description": "devadmin created version FV_4FCEYZ3T of file SF_Y07IN1HO",
              "__typename": "EventNode"
            },
            "__typename": "EventNodeEdge"
          },
          {
            "node": {
              "id": "RXZlbnROb2RlOjMx",
              "uuid": "578c24b2-dcf3-4a80-a4e8-d240e9549736",
              "createdAt": "2019-08-26T18:23:53.695244+00:00",
              "eventType": "SF_CRE",
              "description": "devadmin created file SF_Y07IN1HO",
              "__typename": "EventNode"
            },
            "__typename": "EventNodeEdge"
          }
        ],
        "__typename": "EventNodeConnection"
      },
      "projects": {
        "edges": [
          {
            "node": {
              "id": "UHJvamVjdE5vZGU6emh1eDMvc2QtNTUwNDB4bmY=",
              "createdBy": "zhux3",
              "createdOn": "2019-08-26T18:22:41+00:00",
              "description": "",
              "modifiedOn": "2019-08-26T18:22:41+00:00",
              "name": "test study name",
              "projectId": "zhux3/sd-8wx8qq06",
              "projectType": "DEL",
              "url": "https://cavatica-api.sbgenomics.com/v2/projects/zhux3/sd-8wx8qq06",
              "workflowType": null,
              "deleted": false,
              "__typename": "ProjectNode",
              "study": {
                "id": "U3R1ZHlOb2RlOlNEXzhXWDhRUTA2",
                "kfId": "SD_8WX8QQ06",
                "name": "benchmark extensible e-business",
                "shortName": null,
                "createdAt": "2019-04-20T15:22:18.780221+00:00",
                "modifiedAt": "2019-04-25T15:22:18.780221+00:00",
                "__typename": "StudyNode"
              }
            },
            "__typename": "ProjectNodeEdge"
          },
          {
            "node": {
              "id": "UHJvamVjdE5vZGU6emh1eDMvc2QtNTUwNDB4bmYtYndhLW1lbQ==",
              "createdBy": "zhux3",
              "createdOn": "2019-08-26T18:22:42+00:00",
              "description": "",
              "modifiedOn": "2019-08-26T18:22:42+00:00",
              "name": "test study name bwa-mem",
              "projectId": "zhux3/sd-8wx8qq06-bwa-mem",
              "projectType": "HAR",
              "url": "https://cavatica-api.sbgenomics.com/v2/projects/zhux3/sd-8wx8qq06-bwa-mem",
              "workflowType": "bwa_mem",
              "deleted": false,
              "__typename": "ProjectNode",
              "study": {
                "id": "U3R1ZHlOb2RlOlNEXzhXWDhRUTA2",
                "kfId": "SD_8WX8QQ06",
                "name": "benchmark extensible e-business",
                "shortName": null,
                "createdAt": "2019-04-20T15:22:18.780221+00:00",
                "modifiedAt": "2019-04-25T15:22:18.780221+00:00",
                "__typename": "StudyNode"
              }
            },
            "__typename": "ProjectNodeEdge"
          },
          {
            "node": {
              "id": "UHJvamVjdE5vZGU6emh1eDMvc2QtNTUwNDB4bmYtZ2F0ay1oYXBsb3R5cGVjYWxsZXI=",
              "createdBy": "zhux3",
              "createdOn": "2019-08-26T18:22:43+00:00",
              "description": "",
              "modifiedOn": "2019-08-26T18:22:43+00:00",
              "name": "test study name gatk-haplotypecaller",
              "projectId": "zhux3/sd-8wx8qq06-gatk-haplotypecaller",
              "projectType": "HAR",
              "url": "https://cavatica-api.sbgenomics.com/v2/projects/zhux3/sd-8wx8qq06-gatk-haplotypecaller",
              "workflowType": "gatk_haplotypecaller",
              "deleted": false,
              "__typename": "ProjectNode",
              "study": {
                "id": "U3R1ZHlOb2RlOlNEXzhXWDhRUTA2",
                "kfId": "SD_8WX8QQ06",
                "name": "benchmark extensible e-business",
                "shortName": null,
                "createdAt": "2019-04-20T15:22:18.780221+00:00",
                "modifiedAt": "2019-04-25T15:22:18.780221+00:00",
                "__typename": "StudyNode"
              }
            },
            "__typename": "ProjectNodeEdge"
          }
        ],
        "__typename": "ProjectNodeConnection"
      },
      "__typename": "StudyNode"
    }
  }
}
