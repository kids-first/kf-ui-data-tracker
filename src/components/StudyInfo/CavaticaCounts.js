import React from 'react';
import PropTypes from 'prop-types';
import {Link} from 'react-router-dom';
import {List, Popup, Label} from 'semantic-ui-react';
import {projectOptions} from '../../common/enums';
import CavaticaLogo from '../../assets/CavaticaLogo';
import {withAnalyticsTracking} from '../../analyticsTracking';
/**
 * Displays project counts with total number and breaking down by each type
 * When no projects exist, show buttons guiding user to add/link projects
 */
const CavaticaCounts = ({
  projects,
  title,
  hideIcon,
  tracking: {buttonTracking, popupTracking, logEvent, inheritedEventProps},
}) => {
  const types = projects.map(({node: {projectType}}) => projectType);
  const typeCounts = types.reduce((count, type) => {
    count[type] = (count[type] || 0) + 1;
    return count;
  }, {});
  return (
    <List horizontal>
      <List.Item
        as={Link}
        to={`/study/${title}/cavatica`}
        onClick={e => {
          e.stopPropagation();
          logEvent(
            inheritedEventProps.scope
              ? inheritedEventProps.scope + '__TOOLTIP_PROJECTS__CLICK'
              : 'TOOTLIP_PROJECTS__CLICK',
            {
              tooltip_name: 'Projects',
              tooltip_content: `${
                projects.length > 0 ? projects.length : 'No'
              } Projects`,
              link: `/study/${title}/cavatica`,
            },
          );
        }}
        className={hideIcon && projects.length === 0 ? 'text-red' : null}
        {...buttonTracking({
          button_text: `${
            projects.length > 0 ? projects.length : 'No'
          } projects`,
          button_type: 'text link',
          link: `/study/${title}/cavatica`,
        })}
      >
        {!hideIcon && (
          <CavaticaLogo
            className="mr-5 vertical-middle"
            fill={
              projects && projects.length > 0 ? 'rgba(0,0,0,.6)' : '#db2828'
            }
          />
        )}
        {projects.length > 0 ? projects.length : 'No'} projects
      </List.Item>
      {projectOptions.map(
        type =>
          type.value in typeCounts && (
            <Popup
              inverted
              position="top center"
              size="small"
              content={type.text}
              key={type.key}
              trigger={
                <List.Item
                  {...popupTracking({
                    name: typeCounts[type.key],
                    content: type.text,
                  })}
                >
                  <Label circular empty size="mini" color="olive" />{' '}
                  {typeCounts[type.key]}
                </List.Item>
              }
            />
          ),
      )}
    </List>
  );
};

CavaticaCounts.propTypes = {
  /** Array of project object*/
  projects: PropTypes.array.isRequired,
  /** Study id used by forming the redirect url*/
  title: PropTypes.string.isRequired,
};

CavaticaCounts.defaultProps = {
  projects: [],
};

export default withAnalyticsTracking(CavaticaCounts);
