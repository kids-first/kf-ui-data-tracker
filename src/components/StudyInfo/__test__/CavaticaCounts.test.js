import React from 'react';
import {render, cleanup} from 'react-testing-library';
import {MemoryRouter} from 'react-router-dom';
import CavaticaCounts from '../CavaticaCounts';
import studyByKfId from '../../../../__mocks__/kf-api-study-creator/responses/studyByKfId';

afterEach(cleanup);

it('renders CavaticaCounts in default stage', () => {
  const projects = studyByKfId.data.studyByKfId.projects.edges;
  const studyId = studyByKfId.data.studyByKfId.kfId;
  const tree = render(
    <MemoryRouter>
      <CavaticaCounts projects={projects} title={studyId} />
    </MemoryRouter>,
  );
  expect(tree.container).toMatchSnapshot();
});

it('renders CavaticaCounts with empty project list', () => {
  const studyId = studyByKfId.data.studyByKfId.kfId;
  const tree = render(
    <MemoryRouter>
      <CavaticaCounts title={studyId} />
    </MemoryRouter>,
  );
  expect(tree.container).toMatchSnapshot();
});
