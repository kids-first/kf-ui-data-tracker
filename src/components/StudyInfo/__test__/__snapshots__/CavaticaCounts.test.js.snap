// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders CavaticaCounts in default stage 1`] = `
<div>
  <div
    class="ui horizontal list"
    role="list"
  >
    <a
      class="item"
      href="/study/SD_8WX8QQ06/cavatica"
      role="listitem"
    >
      <i
        aria-hidden="true"
        class="code branch icon"
      />
      3
       projects
    </a>
    <div
      class="item"
      role="listitem"
    >
      <div
        class="ui olive mini circular empty label"
      />
       
      1
    </div>
    <div
      class="item"
      role="listitem"
    >
      <div
        class="ui olive mini circular empty label"
      />
       
      2
    </div>
  </div>
</div>
`;

exports[`renders CavaticaCounts with empty project list 1`] = `
<div>
  <div
    class="ui horizontal list"
    role="list"
  >
    <div
      class="item"
      role="listitem"
    >
      <i
        aria-hidden="true"
        class="red code branch icon"
      />
      <div
        class="ui mini basic compact buttons"
      >
        <a
          class="ui mini primary button"
          href="/study/SD_8WX8QQ06/cavatica#link-cavatica-project"
          role="button"
        >
          <i
            aria-hidden="true"
            class="linkify icon"
          />
          LINK
        </a>
        <a
          class="ui mini basic primary button"
          href="/study/SD_8WX8QQ06/cavatica#add-cavatica-project"
          role="button"
        >
          <i
            aria-hidden="true"
            class="add icon"
          />
          NEW 
        </a>
      </div>
    </div>
  </div>
</div>
`;
